src/Microsoft.Extensions.ProjectModel.Sources/DotNet/DotNetDependencyProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Compilation;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.ProjectModel.Resolution;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class DotNetDependencyProvider
+    {
+        private ProjectContext _context;
+        private List<DependencyDescription> _packageDependencies;
+        private List<ResolvedReference> _resolvedReferences;
+        private string _configuration;
+
+        public DotNetDependencyProvider(ProjectContext context, string configuration = "Debug")
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            _configuration = configuration;
+            _context = context;
+            DiscoverDependencies();
+
+        }
+
+        public IEnumerable<DependencyDescription> GetPackageDependencies()
+        {
+            return _packageDependencies;
+        }
+
+        public IEnumerable<ResolvedReference> GetResolvedReferences()
+        {
+            return _resolvedReferences;
+        }
+
+        private void DiscoverDependencies()
+        {
+            LibraryExporter exporter = _context.CreateExporter(_configuration);
+
+            var framework = _context.TargetFramework;
+            if (exporter == null)
+            {
+                throw new ArgumentNullException(nameof(exporter));
+            }
+
+            if (framework == null)
+            {
+                throw new ArgumentNullException(nameof(framework));
+            }
+
+            var exports = exporter.GetAllExports();
+            var nugetPackages = new Dictionary<string, DependencyDescription>();
+            _resolvedReferences = new List<ResolvedReference>();
+
+            foreach (var export in exports)
+            {
+                var library = export.Library;
+                bool isResolved = bool.TryParse(library.Resolved.ToString(), out isResolved)
+                    ? isResolved
+                    : false;
+
+                var itemSpec = $"{framework.DotNetFrameworkName}/{library.Identity.Name}/{library.Identity.Version.ToString()}";
+                var description = new DependencyDescription(library.Identity.Name,
+                    export.Library.Path,
+                    itemSpec,
+                    library.Identity.Version.ToString(),
+                    library.Identity.Type.Value,
+                    isResolved);
+
+                foreach (var d in export.Library.Dependencies)
+                {
+                    description.AddDependency(new Dependency(d.Name, d.VersionRange.MinVersion.ToString()));
```

Comment:
Not quite sure what the right thing here is. for now, we don't have any scenarios that lookup the version for the `Dependency`
So maybe it would be ok to keep it empty or remove it completely.